MobiusCore/Source/EffectHandlers/EffectRouterDSL.swift

```diff
@@ -0,0 +1,166 @@
+// Copyright (c) 2019 Spotify AB.
+//
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+public extension EffectRouter where Input: Equatable {
+    /// Route to an `EffectHandler` for all effects which are equal to `constant`.
+    /// - Parameter constant: the constant to use for comparisons
+    /// - Parameter handler: the effect handler which should handle `constant`
+    func routeConstant(
```

Comment:
This is distinctly [unSwifty](https://swift.org/documentation/api-design-guidelines/#naming) naming. A more natural name would be `route(constant:to:)` or `route(_:to:)`. I suppose we’ll run into type inference problems if we use `route(_:to:)` for both the predicate and extractor variants, though.

I’d particularly like to avoid `toVoid`, which is redundantly naming a type without providing additional context.

Encoding the entire matrix of criterion/target as distinct functions feels a bit iffy. How do you feel about `.route(criterion).to(target)`?

Reply:
I think the `.route(criterion).to(target)` idea is interesting. I thought it might be too complex for users, and that we should take that pain on the implementation side, but I'll play around with it and see how it feels.

Reply:
Result: yes, it's better. Thanks :)